docs/reference/builder.md

```diff
@@ -1296,11 +1296,16 @@ into the newly created volume.
 
 ## USER
 
-    USER daemon
+    USER daemon[:wheel]
+
+The `USER` instruction sets the user name (or UID) and optionally the user
+group (or GID) to use when running the image and for any `RUN`, `CMD` and
+`ENTRYPOINT` instructions that follow it in the `Dockerfile`.
+
+> **Warning**:
+> When the user group is omited then it's possible that the image (or the next
+> instructions) will be running with the `root` group.
```

Comment:
Perhaps we should be more explicit, something like;

```markdown
> **Warning**:
> IF `USER` is used without specifying a `group` or `GID`, the group is 
> reset, and the default group (`0` - `root`) is used.
```

Steps to check this behavior;

```bash
$ docker build -<<EOF
FROM busybox
RUN id
USER 123:456
RUN id
USER 789
RUN id
EOF
```

Which will show;

```
Sending build context to Docker daemon  2.048kB
Step 1/6 : FROM busybox
 ---> 00f017a8c2a6
Step 2/6 : RUN id
 ---> Running in 1e541467e8a4
uid=0(root) gid=0(root) groups=10(wheel)
 ---> f683b31c30a0
Removing intermediate container 1e541467e8a4
Step 3/6 : USER 123:456
 ---> Running in 20124639421b
 ---> ebcd2f31886f
Removing intermediate container 20124639421b
Step 4/6 : RUN id
 ---> Running in 36079bff5338
uid=123 gid=456
 ---> fa3b4d02cd7f
Removing intermediate container 36079bff5338
Step 5/6 : USER 789
 ---> Running in f54165b19c52
 ---> c4209b5a44fb
Removing intermediate container f54165b19c52
Step 6/6 : RUN id
 ---> Running in 30ae2b70c550
uid=789 gid=0(root)
 ---> b39cfbb318c3
Removing intermediate container 30ae2b70c550
Successfully built b39cfbb318c3
```

So;

- by default, both `uid` and `gid` are `0` (`root`)
- if both `uid` and `gid` are provided, `uid:gid` is used
- if only `uid` is provided, `gid` is reset to the default (`0` - `root`)


This is different than, e.g. `sudo -u`, which preserves the current `gid`;

```bash
$ sudo -u \#123 id
uid=123 gid=20(staff) groups=20(staff),4294967295(nogroup)
```